sourcegraph/sourcegraph
# Code Insights: Add filters section to standalone page (backend insight)

Closes: https://github.com/sourcegraph/sourcegraph/issues/34763

## Background

This PR adds the drill-down panel component (horizontal mode) to the standalone backend insight page. 

<img width="1043" alt="Screenshot 2022-05-06 at 17 16 38" src="https://user-images.githubusercontent.com/18492575/167103728-aca03dc0-1f7c-442e-916a-1e479b6a0ceb.png">

## How to test
- Gather the insight id that you want to look at on the standalone page (you can go to the edit UI of this insight and copy the insight id in URL of edit page
- Go to the `/insights/insight/<id>` URL

## Test plan
- Make sure that filter logic works as expected (both search and repositories filters) 
- Make sure that "save as new view" logic creates a new insights with pre-defined filters 

## Follow up work
In this version of this component this filter view doesn't have a collapsed state (we have this state in our designs about this panel [here](https://www.figma.com/file/Y4grZjTxnXHEQSoeB14vU1/Share-a-link-to-an-individual-code-insight-RFC-654?node-id=1228%3A16270)

Follow up issue: https://github.com/sourcegraph/sourcegraph/issues/35044






## App preview:

- [Web](https://sg-web-vk-add-filters-to-standalone-page.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-mhdixkfczi.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
